### ISSUE TITLE
Print sometimes printing out-of-order when using multiple invocations in succession

### ISSUE THREAD

**Issue by EmilBode** [spyder-ide/spyder#9028]
### Issue Report Checklist

* [X ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [X ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [X ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [X ] Tried basic troubleshooting (if a bug/error)
    * [X ] Restarted Spyder
    * [X ] Reset preferences with ``spyder --reset``
    * [X ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [X ] Tried the other applicable steps from the Troubleshooting Guide
* [X ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
When printing in rapid succession, the IPython console sometimes starts printing a 2nd string before the first one is finished, which can be problematic with end='\r'

### What steps reproduce the problem?

1. Run the following code:
````
from random import random
from time import sleep
for indexcounter in range(1000):
    # Do some stuff of variable duration, e.g.:
    sleep(.1*random()**2)
    print(indexcounter/10, '% completed', end='\r')
````

### What is the expected output? What do you see instead?
I expected a percentage counter, that continously overwrites itself, so in the end we see 99.9 % completed, and nothing more.
Instead, sometimes the print gets pasted at the end, so when finished, I see something like this:
99.9 % completed.4 % completed2.3 % completed4.2% completed, etc.

### Extra information/workaround
- Adding `sys,stdout.flush()` doesn't seem to make a difference
- Adding '\r' to the start of the string as well seems to mitigate, but not solve the problem
- At my setup, I can work around it by prebuilding a string, added with multiple '\r's:
````
printstring='\r'+str(indexcounter/10)+' % completed\r'
print(printstring, end='\r')
````

## Versions

* Spyder version: 3.3.2
* Python version: 3.7.1 (64 bits)
* Qt version: 5.9.3
* PyQt version: PyQt5 5.9.2
* IPython version: 7.2.0
* Operating System name/version: Windows 10 (64 bits)

### Dependencies
```
IPython >=4.0     :  7.2.0 (OK)
cython >=0.21     :  0.29.2 (OK)
jedi >=0.9.0      :  0.13.2 (OK)
matplotlib >=2.0.0:  3.0.2 (OK)
nbconvert >=4.0   :  5.4.0 (OK)
numpy >=1.7       :  1.15.4 (OK)
pandas >=0.13.1   :  0.23.4 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pyflakes >=0.6.0  :  2.0.0 (OK)
pygments >=2.0    :  2.3.1 (OK)
pylint >=0.25     :  2.2.2 (OK)
qtconsole >=4.2.0 :  4.4.3 (OK)
rope >=0.9.4      :  0.11.0 (OK)
sphinx >=0.6.6    :  1.8.2 (OK)
sympy >=0.7.3     :  1.3 (OK)
```

**Comment by ccordoba12** [comment]
Thanks for reporting. We'll take a look at it in a future release.

**Comment by bcolsen** [comment]
This issue occurs in qtconsole aswell

**Comment by ccordoba12** [comment]
This was fixed in Spyder 5.3.3, released a couple of days ago.
